Dockerfile

```diff
@@ -77,30 +84,46 @@ RUN xmlstarlet ed -L \
 # Remove unwanted files from distribution
 RUN rm -fr webapps/* *.txt *.md RELEASE-NOTES logs/ temp/ work/ bin/*.bat
 
+# hadolint ignore=DL3041
 FROM ${IMAGE_JAVA_REPO}/${IMAGE_JAVA_NAME}:${IMAGE_JAVA_TAG} AS tcnative_build-rockylinux
+ARG DISTRIB_MAJOR
 ARG JAVA_MAJOR
 ENV JAVA_HOME=/usr/lib/jvm/java-openjdk
 ARG BUILD_DIR=/build
 ARG INSTALL_DIR=/usr/local
 COPY --from=tomcat_dist /build/tcnative $BUILD_DIR/tcnative
-WORKDIR ${BUILD_DIR}/tcnative/native
+COPY --from=tomcat_dist /build/apr $BUILD_DIR/apr
 SHELL ["/bin/bash", "-euo", "pipefail", "-c"]
-RUN yum install -y gcc make openssl-devel expat-devel java-${JAVA_MAJOR}-openjdk-devel apr-devel redhat-rpm-config && yum clean all; \
-  ./configure \
+RUN yum install -y gcc make expat-devel java-${JAVA_MAJOR}-openjdk-devel redhat-rpm-config && \
+    yum clean all
+WORKDIR ${BUILD_DIR}/apr
+RUN ./configure --prefix=${INSTALL_DIR}/apr; \
+  make -j "$(nproc)"; \
+  make install
+WORKDIR ${BUILD_DIR}/tcnative/native
+RUN if [ $DISTRIB_MAJOR -eq 8 ]; then dnf install -y dnf-plugins-core; \
+    dnf config-manager -y --set-enabled powertools; \
+    dnf install -y https://dl.fedoraproject.org/pub/epel/epel-release-latest-8.noarch.rpm; \
+    dnf install -y openssl3-devel; \
+    ln -s /usr/include/openssl3/openssl /usr/include/openssl; \
+    export LIBS="-L/usr/lib64/openssl3 -Wl,-rpath,/usr/lib64/openssl3 -lssl -lcrypto"; \
+    export CFLAGS="-I/usr/include/openssl3"; \
+    else dnf install -y openssl-devel; \
+    fi; \
```

Comment:
we have multi-line commands not following the same code style, some times we are using `;` and sometimes `&&` - which should not make any difference in this context

it seems there is also a new heredoc notation since recent buildkit
https://docs.docker.com/reference/dockerfile/#here-documents

restrain from changing untouched lines in this pr even if you feel the urge to do so, better do the code style change in a new pr